hail/src/main/scala/is/hail/expr/ir/Compilable.scala

```diff
@@ -60,5 +60,9 @@ object Emittable {
     case _: AggExplode => true
     case _ => false
   }
-  def apply(ir: IR): Boolean = Compilable(ir) && !isNonEmittableAgg(ir)
+  def apply(ir: IR): Boolean = ir match {
+    case x if isNonEmittableAgg(x) => false
+    case _: ApplyIR => false
```

Comment:
Would it make sense to have a isNonEmittable that includes  ApplyIR?

Reply:
I kind of feel like the design of the IRState/Rule stuff in its entirety isn't quite right. 

You'll also be adding streamify on the end of this, which will add its own restrictions. Let's wait until that goes in and then redesign to make things a little simpler.

Reply:
Agreed!